File: iosApp/iosApp/Features/Onboarding/NotifPermissionSheet.swift
import SwiftUI

struct NotifPermissionSheet: View {
    var onDismiss: () -> Void = {}
    var onOpenSettings: () -> Void = {}

    var body: some View {
        VStack(spacing: 16) {
            ZStack {
                Circle()
                    .fill(Color.sgPrimarySoft)
                    .frame(width: 72, height: 72)
                Text("\u{f0f3}")
                    .font(.custom("FontAwesome6Free-Solid", size: 32))
                    .foregroundColor(.sgPrimary)
            }

            Text("Notifikasi dimatikan")
                .font(.custom("PlusJakartaSans-ExtraBold", size: 20))
                .foregroundColor(.sgTextPrimary)

            Text("ServisGo butuh izin notifikasi sistem supaya bisa kirim pengingat servis tepat waktu.")
                .font(.custom("PlusJakartaSans-Medium", size: 14))
                .foregroundColor(.sgTextMuted)
                .multilineTextAlignment(.center)
                .padding(.horizontal, 8)

            VStack(spacing: 10) {
                step(1, "Tap \"Buka Pengaturan\" di bawah")
                step(2, "Pilih \"Notifikasi\" lalu cari ServisGo")
                step(3, "Aktifkan toggle notifikasi utama")
            }
            .padding(16)
            .frame(maxWidth: .infinity, alignment: .leading)
            .background(Color.sgSurfaceAlt)
            .clipShape(RoundedRectangle(cornerRadius: 14))

            Button(action: {
                onOpenSettings()
                onDismiss()
            }) {
                Text("Buka Pengaturan iOS")
                    .font(.custom("PlusJakartaSans-Bold", size: 15))
                    .foregroundColor(.white)
                    .frame(maxWidth: .infinity)
                    .frame(height: 52)
                    .background(Color.sgPrimary)
                    .clipShape(RoundedRectangle(cornerRadius: 14))
            }
            .buttonStyle(.plain)
            .padding(.top, 4)

            Button(action: onDismiss) {
                Text("Nanti saja")
                    .font(.custom("PlusJakartaSans-SemiBold", size: 13))
                    .foregroundColor(.sgTextMuted)
                    .frame(maxWidth: .infinity)
                    .frame(height: 40)
            }
            .buttonStyle(.plain)
        }
        .padding(24)
        .background(Color.sgSurface)
    }

    private func step(_ number: Int, _ text: String) -> some View {
        HStack(alignment: .center, spacing: 12) {
            ZStack {
                Circle()
                    .fill(Color.sgPrimary)
                    .frame(width: 24, height: 24)
                Text("\(number)")
                    .font(.custom("PlusJakartaSans-ExtraBold", size: 12))
                    .foregroundColor(.white)
            }
            Text(text)
                .font(.custom("PlusJakartaSans-Medium", size: 13))
                .foregroundColor(.sgTextPrimary)
            Spacer()
        }
    }
}

File: composeApp/src/androidMain/kotlin/com/zrifapps/goservice/ui/onboarding/NotifPermissionSheet.kt
package com.zrifapps.goservice.ui.onboarding

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.PaddingValues
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.runtime.rememberCoroutineScope
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.zrifapps.goservice.ui.theme.AppColors
import com.zrifapps.goservice.ui.theme.FaIcon
import com.zrifapps.goservice.ui.theme.FaIcons
import com.zrifapps.goservice.ui.theme.plusJakartaSansFontFamily
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun NotifPermissionSheet(
    onDismiss: () -> Unit,
    onOpenSystemSettings: () -> Unit,
) {
    val font = plusJakartaSansFontFamily()
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)
    val scope = rememberCoroutineScope()

    ModalBottomSheet(
        onDismissRequest = onDismiss,
        sheetState = sheetState,
        containerColor = AppColors.Surface,
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp)
                .padding(bottom = 32.dp),
            horizontalAlignment = Alignment.CenterHorizontally,
        ) {
            Box(
                modifier = Modifier
                    .size(72.dp)
                    .clip(CircleShape)
                    .background(AppColors.PrimarySoft),
                contentAlignment = Alignment.Center,
            ) {
                FaIcon(icon = FaIcons.BELL, color = AppColors.Primary, size = 32.sp)
            }
            Spacer(Modifier.height(16.dp))
            Text(
                text = "Notifikasi dimatikan",
                color = AppColors.TextPrimary,
                fontSize = 20.sp,
                fontWeight = FontWeight.ExtraBold,
                fontFamily = font,
                letterSpacing = (-0.3).sp,
            )
            Spacer(Modifier.height(8.dp))
            Text(
                text = "ServisGo butuh izin notifikasi sistem supaya bisa kirim pengingat servis tepat waktu.",
                color = AppColors.TextMuted,
                fontSize = 14.sp,
                fontWeight = FontWeight.Medium,
                fontFamily = font,
                lineHeight = 20.sp,
                modifier = Modifier.padding(horizontal = 8.dp),
                textAlign = androidx.compose.ui.text.style.TextAlign.Center,
            )
            Spacer(Modifier.height(20.dp))

            Column(
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(14.dp))
                    .background(AppColors.SurfaceAlt)
                    .padding(16.dp),
                verticalArrangement = Arrangement.spacedBy(10.dp),
            ) {
                Step(number = 1, text = "Tap \"Buka Pengaturan\" di bawah")
                Step(number = 2, text = "Pilih \"Notifikasi\" lalu cari ServisGo")
                Step(number = 3, text = "Aktifkan toggle notifikasi utama")
            }

            Spacer(Modifier.height(20.dp))

            Box(
                modifier = Modifier
                    .fillMaxWidth()
                    .height(52.dp)
                    .clip(RoundedCornerShape(14.dp))
                    .background(AppColors.Primary)
                    .clickable(onClick = {
                        scope.launch { sheetState.hide() }.invokeOnCompletion {
                            onOpenSystemSettings()
                            onDismiss()
                        }
                    }),
                contentAlignment = Alignment.Center,
            ) {
                Text(
                    text = "Buka Pengaturan",
                    color = Color.White,
                    fontSize = 15.sp,
                    fontWeight = FontWeight.Bold,
                    fontFamily = font,
                )
            }
            Spacer(Modifier.height(6.dp))
            Text(
                text = "Nanti saja",
                color = AppColors.TextMuted,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                fontFamily = font,
                modifier = Modifier
                    .clickable(onClick = onDismiss)
                    .padding(PaddingValues(horizontal = 14.dp, vertical = 10.dp)),
            )
        }
    }
}

@Composable
private fun Step(number: Int, text: String) {
    val font = plusJakartaSansFontFamily()
    Row(
        verticalAlignment = Alignment.CenterVertically,
        horizontalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Box(
            modifier = Modifier
                .size(24.dp)
                .clip(CircleShape)
                .background(AppColors.Primary),
            contentAlignment = Alignment.Center,
        ) {
            Text(
                text = number.toString(),
                color = Color.White,
                fontSize = 12.sp,
                fontWeight = FontWeight.ExtraBold,
                fontFamily = font,
            )
        }
        Text(
            text = text,
            color = AppColors.TextPrimary,
            fontSize = 13.sp,
            fontWeight = FontWeight.Medium,
            fontFamily = font,
            modifier = Modifier.weight(1f),
        )
    }
}
